src/components/pages/tx/transaction-details.tsx

```diff
@@ -1,44 +1,105 @@
 import Link from "next/link";
-import { Clock } from "lucide-react";
-import { formatTimestamp } from "@/lib/utils";
+import { Clock, ChevronRight } from "lucide-react";
+import { formatTimestamp, parseTokenTransfers } from "@/lib/utils";
 import DescriptionListItem from "@/components/lib/description-list-item";
 import EthereumIcon from "@/components/lib/ethereum-icon";
 import { Card, CardContent } from "@/components/ui/card";
 import { Transaction } from "@/lib/types";
-import { formatGwei, formatEther } from "viem";
-
-const TransactionDetails = ({ transaction }: { transaction: Transaction }) => (
-  <Card>
-    <CardContent className="p-4">
-      <dl>
-        <DescriptionListItem title="Transaction Hash">
-          {transaction.hash}
-        </DescriptionListItem>
-        <DescriptionListItem title="Block">
-          <Link
-            className="text-primary hover:brightness-150"
-            href={`/block/${transaction.blockNumber}`}
-          >
-            {transaction.blockNumber.toString()}
-          </Link>
-        </DescriptionListItem>
-        <DescriptionListItem title="Timestamp">
-          <Clock className="mr-1 size-4" />
-          {formatTimestamp(transaction.timestamp)}
-        </DescriptionListItem>
-        <DescriptionListItem border title="Value">
-          <EthereumIcon className="mr-1 size-4" />
-          {formatEther(transaction.value)} ETH
-        </DescriptionListItem>
-        <DescriptionListItem title="Gas Price">
-          {formatGwei(transaction.gasPrice ?? BigInt(0))} Gwei
-          <span className="ml-1 text-muted-foreground">
-            ({formatEther(transaction.gasPrice ?? BigInt(0))} ETH)
-          </span>
-        </DescriptionListItem>
-      </dl>
-    </CardContent>
-  </Card>
-);
+import { formatGwei, formatEther, GetValue} from "viem";
+import { l2PublicClient } from "@/lib/chains";
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "@/components/ui/tooltip";
 
+const TransactionDetails = async ({ transaction }: { transaction: Transaction }) => {
+  const receipt = await l2PublicClient.getTransactionReceipt({ hash: transaction.hash });
```

Comment:
This could be moved one level up in the top-level Page component and await-ed with the `getTransaction` call using a `Promise.all`.
Then the `parseTokenTransfers` can be called and you could enrich the result by adding a `decimals`
 property, by calling the underlying token address `decimals` view function.
You will need to add a new simple ERC20 contract under `@/lib/contracts/erc-20` to be able to call the view function properly.